src/jaxsim/rbda/contacts/relaxed_rigid.py

```diff
@@ -376,7 +376,7 @@ def compute_contact_forces(
 
         # Compute the Delassus matrix and the free mixed linear acceleration of
         # the collidable points.
-        G = Jl_WC @ jnp.linalg.lstsq(M, Jl_WC.T)[0]
+        G = Jl_WC @ jnp.linalg.inv(M) @ Jl_WC.T
```

Comment:
Did you check this change when the free-floating mass matrix $M$ corresponds to almost singular configurations? How comes that by using the plain inverse, the tolerance in the test is now met? I fear that this change might introduce problem on more complex simulations.

Reply:
Yes, I also fear that this could bring some numerical issues in singular configurations.. 
Have you tried to maybe use the `rcond` parameter to see if the issue improves? Or to use [pinv](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.linalg.pinv.html)?

Reply:
I'd go with `pinv` instead of `inv`. The result should be the same as `lstsq`, if it's faster, much better :)

Reply:
Started investigation in https://github.com/ami-iit/jaxsim/pull/295